devtools/aesh/src/main/java/io/quarkus/cli/commands/CompileProjectCommand.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkus.cli.commands;
+
+import java.io.File;
+
+import org.aesh.command.Command;
+import org.aesh.command.CommandDefinition;
+import org.aesh.command.CommandException;
+import org.aesh.command.CommandResult;
+import org.aesh.command.invocation.CommandInvocation;
+import org.aesh.command.option.Argument;
+import org.aesh.command.option.Option;
+import org.aesh.io.Resource;
+
+import io.quarkus.devtools.project.BuildTool;
+import io.quarkus.devtools.project.QuarkusProject;
+
+@CommandDefinition(name = "compile-project", description = "Compiles the targeted project")
+public class CompileProjectCommand implements Command<CommandInvocation> {
+    @Option(name = "clean", hasValue = false, shortName = 'c', description = "Clean the project before compiling")
+    private boolean clean;
+
+    @Argument(description = "Path to the project, if not set it will use the current working directory")
+    private Resource path;
+
+    @Override
+    public CommandResult execute(CommandInvocation invocation) throws CommandException, InterruptedException {
+
+        File projectPath = path != null ? new File(path.getAbsolutePath()) : new File(System.getProperty("user.dir"));
+
+        BuildTool buildTool = QuarkusProject.resolveExistingProjectBuildTool(projectPath.toPath());
+
+        if (buildTool.getBuildFiles() != null && buildTool.getBuildFiles().length > 0) {
+            File buildFile = new File(buildTool.getBuildFiles()[0]);
+
+            if (!buildFile.isFile()) {
+                invocation.println("Was not able to find a build file in: " + projectPath);
+                return CommandResult.FAILURE;
+            }
+
+            try {
+                if (buildTool.equals(BuildTool.MAVEN)) {
+                    File wrapper = ExecuteUtil.getMavenWrapper(projectPath.getAbsolutePath());
+                    if (wrapper != null) {
+                        ExecuteUtil.executeWrapper(invocation, wrapper, "install");
```

Comment:
Not critical now but I would say package rather than install would be done here.

Reply:
Fixed